addon/-private/class/modifier-manager.ts

```diff
@@ -10,16 +12,40 @@ function destroyModifier(modifier: ClassBasedModifier): void {
   modifier.willDestroy();
 }
 
+type CreateModifierArgs313 = [
+  { owner: unknown; class: typeof ClassBasedModifier },
+  ModifierArgs
+];
+type CreateModifierArgs322 = [typeof ClassBasedModifier, ModifierArgs];
+type CreateModifierArgs = CreateModifierArgs313 | CreateModifierArgs322;
+
 export default class ClassBasedModifierManager {
-  capabilities = capabilities('3.13');
+  capabilities = capabilities(gte('3.22.0') ? '3.22' : '3.13');
 
   constructor(private owner: unknown) {}
 
   createModifier(
-    factory: { owner: unknown; class: typeof ClassBasedModifier },
-    args: ModifierArgs
+    ...createModifierArgs: CreateModifierArgs
   ): ClassBasedModifier {
-    const Modifier = factory.class;
+    const [factoryOrClass, args] = createModifierArgs;
+
+    let Modifier: typeof ClassBasedModifier;
+
+    if (gte('3.22.0')) {
+      assert(
+        'expected factory to be a class reference',
+        !('class' in factoryOrClass)
+      );
```

Comment:
I don't fully grok this assertion. What is the point of it? It seems perfectly reasonable to have a Modifier that looks like:

```js
export default class extends Modifier {
  static get class() {
    // stuff!
  }
}
```

Its kinda dumb looking, but it isn't invalid.

---

These two assertions seem like "testing the framework" to me, and I think they could be removed..